src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java

```diff
@@ -34,21 +34,20 @@ public TrustCorporateValidator(AddressDtoValidator addressDtoValidator) {
         this.addressDtoValidator = addressDtoValidator;
     }
 
-    public Errors validate(List<TrustDataDto> trustDataDtoList, Errors errors, String loggingContext) {
+    public Errors validate(List<TrustDataDto> trustDataDtoList, Errors errors, String loggingContext, boolean isForUpdateOrRemove) {
         for (TrustDataDto trustDataDto : trustDataDtoList) {
             List<TrustCorporateDto> corporates = trustDataDto.getCorporates();
-
             if (!CollectionUtils.isEmpty(corporates)) {
                 for (TrustCorporateDto trustCorporateDto : corporates) {
-                    validateTrustCorporateDto(trustCorporateDto, errors, loggingContext);
+                    validateTrustCorporateDto(trustCorporateDto, trustDataDto.getCreationDate(), errors, loggingContext, isForUpdateOrRemove);
                 }
             }
         }
         return errors;
     }
 
-    private void validateTrustCorporateDto(TrustCorporateDto trustCorporateDto, Errors errors,
-            String loggingContext) {
+    private void validateTrustCorporateDto(TrustCorporateDto trustCorporateDto, LocalDate trustCreationDate, Errors errors,
```

Comment:
It might be preferable to keep errors and loggingContext as the last parameters and keep the essential parameters at the beginning.

private void validateTrustCorporateDto(TrustCorporateDto trustCorporateDto, LocalDate trustCreationDate, boolean isForUpdateOrRemove, Errors errors, String loggingContext) { ...

Reply:
Updated as above

Reply:
you have only moved the parameter one position 

errors and logginContext should always come last, as they are not the main parameters for the logic